src/main/java/org/reaktivity/nukleus/tls/internal/stream/ClientStreamFactory.java

```diff
@@ -1215,19 +1181,18 @@ private void handleWindow(
                 }
             }
 
-            final int networkWindowBytesDelta = applicationWindowBytesDelta + networkWindowBytesAdjustment;
-            final int networkWindowFramesDelta = applicationWindowFramesDelta + networkWindowFramesAdjustment;
+            final int networkWindowCredit = window.credit() + networkWindowBudgetAdjustment;
+            final int networkWindowPadding = window.padding();
 
-            networkWindowBytes += Math.max(networkWindowBytesDelta, 0);
-            networkWindowBytesAdjustment = Math.min(networkWindowBytesDelta, 0);
+            networkWindowBudget += Math.max(networkWindowCredit, 0);
+            networkWindowBudgetAdjustment = Math.min(networkWindowCredit, 0);
```

Comment:
`networkWindowCredit` always positive so no need for `Math.max` or `Math.min`, right?

Reply:
networkWindowCredit is negative sometimes because of negative networkWindowBudgetAdjustment